packages/element-lib/src/sidetree/op/index.js

```diff
@@ -117,6 +127,179 @@ const getDeletePayload = async (didUniqueSuffix, recoveryPrivateKey) => {
   return makeSignedOperation(header, payload, recoveryPrivateKey);
 };
 
+const walletToInitionalDIDDoc = wallet => {
+  const didDocumentModel = {
+    '@context': [
+      'https://www.w3.org/ns/did/v1',
+      'https://docs.element-did.com/contexts/sidetree/sidetree-v0.1.jsonld',
+    ],
+  };
+
+  const publicKeys = [];
+  const commonVerificationMethods = [];
+  const keyAgreementkeys = [];
+
+  Object.values(wallet.keys).forEach(walletKey => {
+    if (walletKey.type === 'assymetric') {
+      if (walletKey.tags[0] === 'X25519KeyAgreementKey2019') {
+        keyAgreementkeys.push({
+          id: walletKey.tags[1] || `#${walletKey.kid}`,
+          type: 'X25519KeyAgreementKey2019',
+          usage: 'signing',
+          [walletKey.didPublicKeyEncoding]: walletKey.publicKey,
+        });
+      } else {
+        if (walletKey.encoding === 'jwk') {
+          publicKeys.push({
+            id: walletKey.tags[1] || `#${walletKey.kid}`,
+            type: walletKey.tags[0],
+            usage: 'signing',
+            [walletKey.didPublicKeyEncoding]: JSON.parse(walletKey.publicKey),
+          });
+        } else {
+          publicKeys.push({
+            id:
+              walletKey.tags[1] !== undefined
+                ? walletKey.tags[1]
+                : `#${walletKey.kid}`,
+            type: walletKey.tags[0],
+            usage:
+              walletKey.tags[1] &&
+              walletKey.tags[1].split('#').pop() === 'recovery'
+                ? 'recovery'
+                : 'signing',
+            [walletKey.didPublicKeyEncoding]: walletKey.publicKey,
+          });
+        }
+
+        commonVerificationMethods.push(`#${walletKey.kid}`);
```

Comment:
This is not correct. See this did:
```json
{
  "authentication": [
    "#a4VSF7tSJ7VKSOwkI98m3OHz0OHGIdS5DB_b4fj4gGM",
    "#NFqToUusDIZ3dv1I0xWjWXQFFGqstQK2hntYISIVJpo",
    "#_vLQrqsD3zuRvkVdM0du2Q-CaZPhWWsgxjqO8VPmpoE",
    "#AwE8YZ7HcJ2zORh_XbrDPTFlHTAeNZkLKSrnxYKLnf0"
  ],
  "capabilityDelegation": [
    "#a4VSF7tSJ7VKSOwkI98m3OHz0OHGIdS5DB_b4fj4gGM",
    "#NFqToUusDIZ3dv1I0xWjWXQFFGqstQK2hntYISIVJpo",
    "#_vLQrqsD3zuRvkVdM0du2Q-CaZPhWWsgxjqO8VPmpoE",
    "#AwE8YZ7HcJ2zORh_XbrDPTFlHTAeNZkLKSrnxYKLnf0"
  ],
  "keyAgreement": [
    {
      "type": "X25519KeyAgreementKey2019",
      "publicKeyBase58": "3TkMWjeZvnFtJ1wQZTaSWSruGqsU5GZmRLdJ1YcqKzSp",
      "id": "#keyAgreement",
      "usage": "signing",
      "controller": "did:elem:EiBiiyNodE1Kb3ozmMXp1o9pe_s6Yg-UYUAC4Vh_ZUy6mg"
    }
  ],
  "assertionMethod": [
    "#a4VSF7tSJ7VKSOwkI98m3OHz0OHGIdS5DB_b4fj4gGM",
    "#NFqToUusDIZ3dv1I0xWjWXQFFGqstQK2hntYISIVJpo",
    "#_vLQrqsD3zuRvkVdM0du2Q-CaZPhWWsgxjqO8VPmpoE",
    "#AwE8YZ7HcJ2zORh_XbrDPTFlHTAeNZkLKSrnxYKLnf0"
  ],
  "@context": [
    "https://www.w3.org/ns/did/v1",
    "https://docs.element-did.com/contexts/sidetree/sidetree-v0.1.jsonld"
  ],
  "publicKey": [
    {
      "publicKeyJwk": {
        "x": "HfPpOQ9Qvs3OaQG4-mxpAkAv0HAL2IB_xQP_SRvZuEA",
        "kty": "EC",
        "crv": "secp256k1",
        "y": "-luk_H74kMnty-ko5nIfMqbiw9BxbcRGFijJH4GWWmQ",
        "kid": "a4VSF7tSJ7VKSOwkI98m3OHz0OHGIdS5DB_b4fj4gGM"
      },
      "type": "EcdsaSecp256k1VerificationKey2019",
      "id": "#a4VSF7tSJ7VKSOwkI98m3OHz0OHGIdS5DB_b4fj4gGM",
      "usage": "signing",
      "controller": "did:elem:EiBiiyNodE1Kb3ozmMXp1o9pe_s6Yg-UYUAC4Vh_ZUy6mg"
    },
    {
      "type": "EcdsaSecp256k1VerificationKey2019",
      "id": "#primary",
      "usage": "signing",
      "publicKeyHex": "021df3e9390f50becdce6901b8fa6c6902402fd0700bd8807fc503ff491bd9b840",
      "controller": "did:elem:EiBiiyNodE1Kb3ozmMXp1o9pe_s6Yg-UYUAC4Vh_ZUy6mg"
    },
    {
      "publicKeyHex": "0305347588cb15c7d7cb10ffd94ec0abc1f98428209447b6644a7bced1197ff603",
      "type": "EcdsaSecp256k1VerificationKey2019",
      "id": "#recovery",
      "usage": "recovery",
      "controller": "did:elem:EiBiiyNodE1Kb3ozmMXp1o9pe_s6Yg-UYUAC4Vh_ZUy6mg"
    },
    {
      "type": "Ed25519VerificationKey2018",
      "publicKeyBase58": "CzcfYNghXxbjaYNfqvf8wPMS4va3cLHGXaoUdxZd1DWC",
      "id": "#AwE8YZ7HcJ2zORh_XbrDPTFlHTAeNZkLKSrnxYKLnf0",
      "usage": "signing",
      "controller": "did:elem:EiBiiyNodE1Kb3ozmMXp1o9pe_s6Yg-UYUAC4Vh_ZUy6mg"
    }
  ],
  "capabilityInvocation": [
    "#a4VSF7tSJ7VKSOwkI98m3OHz0OHGIdS5DB_b4fj4gGM",
    "#NFqToUusDIZ3dv1I0xWjWXQFFGqstQK2hntYISIVJpo",
    "#_vLQrqsD3zuRvkVdM0du2Q-CaZPhWWsgxjqO8VPmpoE",
    "#AwE8YZ7HcJ2zORh_XbrDPTFlHTAeNZkLKSrnxYKLnf0"
  ],
  "id": "did:elem:EiBiiyNodE1Kb3ozmMXp1o9pe_s6Yg-UYUAC4Vh_ZUy6mg"
}
```

only the first capabilityInvocation: #a4VSF7tSJ7VKSOwkI98m3OHz0OHGIdS5DB_b4fj4gGM is a valid reference to a key in the document

Reply:
great catch...